thrift/thrift-gen/template.go

```diff
@@ -29,14 +29,42 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os/exec"
+	"strings"
 	"text/template"
 )
 
-func parseTemplate() *template.Template {
+// Template represents a single thrift-gen template that will be used to generate code.
+type Template struct {
+	name     string
+	template *template.Template
+}
+
+func parseTemplate(contents string) (*template.Template, error) {
 	funcs := map[string]interface{}{
-		"contextType": contextType,
+		"contextType":  contextType,
+		"lowercase":    strings.ToLower,
```

Comment:
I was thinking to maybe change this to `goPrivateName` since that is what it is used for and more in line with `goPublicName`. Shows the intent a bit better in templates.